scripts/makepeds_psana

```diff
@@ -374,18 +384,165 @@ printf -v RUNSTR "%04g" $RUN
 if [[ $HOSTNAME =~ "drp-srcf" ]]; then
     echo check FFB files: 
     ls -ltr /cds/data/drpsrcf/$HUTCH/$EXP/xtc/*r$RUNSTR*s*xtc*
-    detnames exp=${EXP}:run=${RUN}:dir=/cds/data/drpsrcf/$HUTCH/$EXP/xtc:stream=0-79 > /tmp/detnames_$EXP\_$CREATE_TIME
+    if [[ $LCLS2 -gt 0 ]]; then
+        detnames -r exp=${EXP},run=${RUN},dir=/cds/data/drpsrcf/$HUTCH/$EXP/xtc > /tmp/detnames_$EXP\_$CREATE_TIME
+    else
+        detnames exp=${EXP}:run=${RUN}:dir=/cds/data/drpsrcf/$HUTCH/$EXP/xtc:stream=0-79 > /tmp/detnames_$EXP\_$CREATE_TIME
+    fi
 else
-    echo Process files: 
+    echo Process offline files: 
     ls -ltr /reg/d/psdm/$HUTCH/$EXP/xtc/*r*$RUNSTR*s*xtc*
-    detnames exp=${EXP}:run=${RUN}:stream=0-79 > /tmp/detnames_$EXP\_$CREATE_TIME
+    if [[ $LCLS2 -gt 0 ]]; then
+        detnames -r exp=${EXP},run=${RUN} > /tmp/detnames_$EXP\_$CREATE_TIME
+    else
+        detnames exp=${EXP}:run=${RUN}:stream=0-79 > /tmp/detnames_$EXP\_$CREATE_TIME
+    fi
 fi
 #running local
 echo -----------------------
 echo 'XTC files contain:'
 cat /tmp/detnames_$EXP\_$CREATE_TIME
 echo -----------------------
 
+#### 
+# Section for LCLS2 detectors.
+####
+if [[ $LCLS2 -gt 0 ]]; then
+    ####
+    # epix100 in LCLS2 related stuff. 
+    ####
+    #check if epix100 detectors are present
+    HAVE_EPIX100=`grep epix100 /tmp/detnames_$EXP\_$CREATE_TIME | wc -l`
+    if [[ $HAVE_EPIX100 -ge 1 ]]; then
+        DETNAMES=`grep epix100 /tmp/detnames_$EXP\_$CREATE_TIME | grep raw | awk  'BEGIN { FS = "|"}; {print $1}' | paste -d " " -s`
+        DETTYPES=`grep epix100 /tmp/detnames_$EXP\_$CREATE_TIME | grep raw | awk  'BEGIN { FS = "|"}; {print $2}' | paste -d " " -s`
+        for i in "${!DETNAMES[@]}"; do
+	    LOCARG='' #optional cuts, I suspect this will not be developed for LV17.
+	    MYDET=${DETNAMES[i]}
+	    MYDETTYPE=${DETTYPES[i]}
+	    echo $MYDETTYPE $MYDET
+	    if [ $MYDETTYPE = 'epix100' ]; then
+                echo 'now calibrate...'
+                cmd=`echo det_dark_proc -r $RUN  -d $MYDET -D -e $EXP $LOCARG`
+        	echo $cmd
+    	        #$cmd
```

Comment:
Nah, that part has not been tested yet - was meant for LV17 and we won't have this type of detector elsewhere. I did not want to test on a running user experiment. I left it in so that I could test it at a better time